packages/flutter_tools/lib/src/windows/visual_studio_validator.dart

```diff
@@ -10,65 +12,73 @@ import 'visual_studio.dart';
 VisualStudioValidator get visualStudioValidator => context.get<VisualStudioValidator>();
 
 class VisualStudioValidator extends DoctorValidator {
-  const VisualStudioValidator() : super('Visual Studio - develop for Windows');
+  const VisualStudioValidator({
+    @required VisualStudio visualStudio,
+    @required UserMessages userMessages,
+  }) : _visualStudio = visualStudio,
+       _userMessages = userMessages,
+       super('Visual Studio - develop for Windows');
+
+  final VisualStudio _visualStudio;
+  final UserMessages _userMessages;
 
   @override
   Future<ValidationResult> validate() async {
     final List<ValidationMessage> messages = <ValidationMessage>[];
     ValidationType status = ValidationType.missing;
     String versionInfo;
 
-    if (visualStudio.isInstalled) {
+    if (_visualStudio.isInstalled) {
       status = ValidationType.installed;
 
       messages.add(ValidationMessage(
-          userMessages.visualStudioLocation(visualStudio.installLocation)
+          _userMessages.visualStudioLocation(_visualStudio.installLocation)
```

Comment:
It seems like there's a potential foot-gun in having this injection while `userMessages` is still visible, since someone could easily use the wrong one without noticing, resulting in hard-to-find issues when injecting. How about `hide`ing userMessages on the import?